src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,118 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TextUi;
+import duke.task.TaskList;
+
+import java.io.IOException;
+
+/**
+ * The Command class is used to pass commands around the Duke application
+ * once user input has been parsed.
+ */
+public abstract class Command {
+    /** Details of the command */
+    protected String details;
+
+    /** Whether the command is an exit command */
+    protected boolean isExit;
+
+    /** Whether the command marks a task as done */
+    protected boolean isDone;
+
+    /**
+     * Constructs a Command with the given details. By default, the boolean
+     * values isExit and isDone are both set to false.
+     *
+     * @param details   Details of the command
+     */
+    Command(String details) {
+        this.details = details;
+        this.isExit = false;
+        this.isDone = false;
```

Comment:
isExit and isDone values can be assigned in the attribute field instead of the constructor.